### ISSUE TITLE
Logstash gets killed because of an exception in loggly output

### ISSUE THREAD

**Issue by vanga** [logstash-plugins/logstash-output-loggly#7]
On ArchLinux, AWS EC2
logstash-1.5.0.rc2
loggly output was cloned on Apr 28.
I have a systemd unit file to run logstash.

```
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: Errno::ECONNREFUSED: Connection refused - Connection refused
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: initialize at org/jruby/ext/socket/RubyTCPSocket.java:126
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: open at org/jruby/RubyIO.java:1177
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: connect at /usr/local/logstash-1.5.0.rc2/vendor/jruby/lib/ruby/1.9/net/http.rb:763
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: timeout at org/jruby/ext/timeout/Timeout.java:104
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: connect at /usr/local/logstash-1.5.0.rc2/vendor/jruby/lib/ruby/1.9/net/http.rb:763
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: do_start at /usr/local/logstash-1.5.0.rc2/vendor/jruby/lib/ruby/1.9/net/http.rb:756
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: start at /usr/local/logstash-1.5.0.rc2/vendor/jruby/lib/ruby/1.9/net/http.rb:745
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: start at /usr/local/logstash-1.5.0.rc2/vendor/jruby/lib/ruby/1.9/net/http.rb:744
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: request at /usr/local/logstash-1.5.0.rc2/vendor/jruby/lib/ruby/1.9/net/http.rb:1293
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: receive at /usr/local/logstash-plugins/logstash-output-loggly/lib/logstash/outputs/loggly.rb:
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: handle at /usr/local/logstash-1.5.0.rc2/lib/logstash/outputs/base.rb:88
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: initialize at (eval):310
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: each at org/jruby/RubyArray.java:1613
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: flat_map at org/jruby/RubyEnumerable.java:805
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: initialize at (eval):298
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: call at org/jruby/RubyProc.java:271
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: output at /usr/local/logstash-1.5.0.rc2/lib/logstash/pipeline.rb:279
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: outputworker at /usr/local/logstash-1.5.0.rc2/lib/logstash/pipeline.rb:235
May 11 00:09:02 ip-172-31-17-141 logstash[12620]: start_outputs at /usr/local/logstash-1.5.0.rc2/lib/logstash/pipeline.rb:163
May 11 00:09:02 ip-172-31-17-141 systemd[1]: logstash.service: main process exited, code=exited, status=1/FAILURE
May 11 00:09:02 ip-172-31-17-141 systemd[1]: Unit logstash.service entered failed state.
May 11 00:09:02 ip-172-31-17-141 systemd[1]: logstash.service holdoff time over, scheduling restart.
```
logstash conf
```
input {
  heartbeat {
    type => "heartbeat"
  }
  s3 {
    bucket => "xxx"
    credentials => [ "xxx", "yyy" ]
    type => "s3-access-logs"
    prefix => "production"
    region_endpoint => "us-west-1"
    region => "us-west-1"
    sincedb_path => "/root/.sincedb_s3-access-logs"
  }
s3 {
    bucket => "xxx"
    credentials => [ "xxx", "yyy" ]
    type => "s3-access-logs-staging"
    prefix => "staging"
    region_endpoint => "us-west-1"
    region => "us-west-1"
    sincedb_path => "/root/.sincedb_s3-access-logs-staging"
  }
  s3 {
    bucket => "xxx"
    credentials => [ "xxx", "yyy" ]
    type => "elb-access-logs"
    region_endpoint => "us-west-1"
    region => "us-west-1"
    sincedb_path => "/root/.sincedb_elb-media"
  }
  s3 {
    bucket => "xxx"
    credentials => [ "xxx", "yyy" ]
    type => "elb-access-logs"
    region_endpoint => "us-west-1"
    region => "us-west-1"
    sincedb_path => "/root/.sincedb_elb-profile"
  }
}
filter {
    if [type] == "s3-access-logs" {
        grok {
            patterns_dir => ["/usr/local/logstash-1.4.2/patterns/"]
            match => { "message" => "%{S3_ACCESS_LOG}" }
            remove_field => ["message"]
            add_field => { "unique_view" => "%{key}_%{clientip}"}
        }
        date {
            match => [ "timestamp" , "dd/MMM/yyyy:HH:mm:ss Z" ]
        }
    }else if [type] == "s3-access-logs-staging" {
        grok {
            patterns_dir => ["/usr/local/logstash-1.4.2/patterns/"]
            match => { "message" => "%{S3_ACCESS_LOG}" }
            remove_field => ["message"]
            add_field => { "unique_view" => "%{key}_%{clientip}"}
        }
        date {
            match => [ "timestamp" , "dd/MMM/yyyy:HH:mm:ss Z" ]
        }
    }else if [type] == "elb-access-logs" {
        grok {
            patterns_dir => ["/usr/local/logstash-1.4.2/patterns/"]
            match => { "message" => "%{ELB_ACCESS_LOG}" }
            remove_field => ["message", "port", "httpversion", "clientport"]
        }
        date {
            match => [ "timestamp" , "ISO8601" ]
        }
    }
}
output {
    if [type] == "s3-access-logs" {
	    elasticsearch { 
	      host => "127.0.0.1"
	      protocol => "http"
	      index => "production-s3-%{+YYYY.ww}"
	    }
    }else if [type] == "s3-access-logs-staging" {
	    elasticsearch { 
	      host => "127.0.0.1"
	      protocol => "http"
	      index => "staging-s3-%{+YYYY.MM}"
	    }
    }else if [type] == "elb-access-logs" {
	    elasticsearch { 
	      host => "127.0.0.1"
	      protocol => "http"
	      index => "production-elb-%{+YYYY.MM}"
	    }
    }else if [type] == "heartbeat" {
       loggly {
        host => "logs-01.loggly.com"
        key => "xxx"
        proto => "https"
      }
    }
}
```

**Comment by psquickitjayant** [comment]
@jordansissel  this issue can be closed as this is addressed in #9

**Comment by jordansissel** [comment]
@psquickitjayant ooh thanks for checking on that. :heart:
